src/runtime/converter.cs

```diff
@@ -758,7 +758,10 @@ private static bool ToPrimitive(IntPtr value, Type obType, out object result, bo
                         goto type_error;
                     }
                     double dd = Runtime.PyFloat_AsDouble(op);
-                    Runtime.CheckExceptionOccurred();
+                    if (dd == -1.0)
```

Comment:
Knowing the funny behavior of floating-point comparisons, it would be good to have two test cases for this conversion passing and failing.